web/src/components/pages/contribution/speak/speak.tsx

```diff
@@ -475,17 +494,61 @@ class SpeakPage extends React.Component<Props, State> {
 
   private agreeToTerms = async () => {
     this.setState({ showPrivacyModal: false });
+    this.setState({ privacyAgreedChecked: true });
     this.props.updateUser({ privacyAgreed: true });
     this.upload(true);
   };
 
+  private onPrivacyAgreedChange = (privacyAgreed: boolean) => {
+    this.setState({ privacyAgreedChecked: privacyAgreed });
+  };
+
   private toggleDiscardModal = () => {
     this.setState({
       showPrivacyModal: false,
       showDiscardModal: !this.state.showDiscardModal,
     });
   };
 
+  private handleSubmit = (evt: React.SyntheticEvent) => {
+    const { user } = this.props;
+
+    const hasSeenFirstCTA = window.sessionStorage.getItem(SEEN_FIRST_CTA);
+    const hasSeenSecondCTA = window.sessionStorage.getItem(SEEN_SECOND_CTA);
+
+    this.props.updateUser({ privacyAgreed: this.state.privacyAgreedChecked });
+
+    evt.preventDefault();
+    this.upload(this.state.privacyAgreedChecked);
+
+    // display first CTA screen if it has not been seen it before
+    // and the user does not have an account
+    if (hasSeenFirstCTA !== 'true' && !user.account) {
+      this.setState({ shouldShowFirstCTA: true });
+      window.sessionStorage.setItem(SEEN_FIRST_CTA, 'true');
+    }
+
+    // display second CTA screen if it has not been seen it before
+    // and the user does not have an account
+    if (
+      hasSeenFirstCTA === 'true' &&
```

Comment:
Question: is this condition checking for the string true or the boolean true?

Reply:
It's checking for the string true. You can only write string values to Session Storage

Reply:
Mixing the type and meaning of 'true' and true is kinda confusing, it might be worth casting the string 'true' into a real boolean once, and then the conditions  will look like all the other ones